engine/src/main/java/org/terasology/rendering/dag/AbstractNode.java

```diff
@@ -15,25 +15,43 @@
  */
 package org.terasology.rendering.dag;
 
-import com.google.common.collect.Lists;
-import com.google.common.collect.Range;
 import com.google.common.collect.Sets;
-
-import java.util.List;
 import java.util.Set;
+import org.terasology.registry.In;
+import org.terasology.rendering.opengl.FBOConfig;
+import org.terasology.rendering.opengl.FrameBuffersManager;
 
 /**
  * TODO: Add javadocs
  */
 public abstract class AbstractNode implements Node {
+    @In
+    private FrameBuffersManager frameBuffersManager;
+
     private Set<StateChange> desiredStateChanges;
+    private Set<String> fboNames;
+
     private NodeTask task;
     private RenderTaskListGenerator taskListGenerator;
-    private List<Range<Integer>> rangeList;
 
     protected AbstractNode() {
         desiredStateChanges = Sets.newLinkedHashSet();
-        rangeList = Lists.newArrayList();
+        fboNames = Sets.newHashSet();
+    }
+
+    protected void requireFBO(FBOConfig fboConfig) {
+
+        if (!frameBuffersManager.isFBOConfigAvailable(fboConfig.getTitle())) {
+            frameBuffersManager.generateFBOWith(fboConfig);
+        }
+        fboNames.add(fboConfig.toString());
+    }
+
+    @Override
+    public void dispose() {
+        for (String fboName : fboNames) {
+            frameBuffersManager.removeUsage(fboName); // TODO: better naming?
```

Comment:
Might this be more about `removeUsage` -> `release` ? "Release" being the opposite of "acquire".